From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Wed, 14 Aug 2024 21:35:16 +0000
Subject: rockchip: rk3566-orangepi-3b: enable npu regulator

The power-domain driver in Linux does not know what regulator is
supplying power to a specific power-domain. This prevent use of NPU with
vendor kernel because vdd_npu is disabled.

Change vdd_npu to use always-on/boot-on to enable the regulator at boot
and set initial 0v9 voltage, the recommended voltage in soc datasheet.

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
---
 arch/arm/dts/rk3566-orangepi-3b-u-boot.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/dts/rk3566-orangepi-3b-u-boot.dtsi b/arch/arm/dts/rk3566-orangepi-3b-u-boot.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm/dts/rk3566-orangepi-3b-u-boot.dtsi
+++ b/arch/arm/dts/rk3566-orangepi-3b-u-boot.dtsi
@@ -12,3 +12,9 @@
 		bootph-some-ram;
 	};
 };
+
+&vdd_npu {
+	regulator-always-on;
+	regulator-boot-on;
+	regulator-init-microvolt = <900000>;
+};
-- 
Armbian

